pkgs/by-name/wh/why3find/package.nix

```
@@ -1,32 +1,49 @@
 {
   lib,
-  fetchurl,
+  stdenv,
+  fetchFromGitLab,
   ocamlPackages,
+  alt-ergo,
+  darwin,
   why3,
 }:
 
-with ocamlPackages;
-buildDunePackage {
+ocamlPackages.buildDunePackage (finalAttrs: {
   pname = "why3find";
   version = "1.2.0";
 
-  src = fetchurl {
-    url = "https://git.frama-c.com/-/project/1056/uploads/043312a7a70961338479016ac535c706/why3find-1.2.0.tbz";
-    hash = "sha256-eslkMBo0i0+Oy8jW9eRNuyGXuwkV6eeYcxZm5MfgA6w=";
+  src = fetchFromGitLab {
+    domain = "git.frama-c.com";
+    owner = "pub";
+    repo = "why3find";
+    tag = finalAttrs.version;
+    hash = "sha256-fqB6VrJ79E6KSAnq8TZGNFlvWbDaBNh+gbuRf0CFFy8=";
   };
 
-  propagatedBuildInputs = [
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
+    darwin.sigtool # codesign
+  ];
+
+  propagatedBuildInputs = with ocamlPackages; [
     dune-site
     terminal_size
     why3
     yojson
     zmq
   ];
 
+  doCheck = true;
+
+  nativeCheckInputs = [
+    alt-ergo
```

```suggestion
    alt-ergo.bin
```

This might save bandwidth.